### PR TITLE
Gpgme and dependencies

### PR DESCRIPTION
**Pull request by jeroenooms**
- Update libgpgme to the latest version. 
 - Enable static libraries for the dependencies of libgpgme, so that we can actually use the libgpgme static library.

### CODE REVIEW

File: Library/Formula/libassuan.rb [hunk 1]
```diff
@@ -18,7 +18,8 @@ class Libassuan < Formula
   def install
     system "./configure", "--disable-dependency-tracking",
                           "--disable-silent-rules",
-                          "--prefix=#{prefix}"
+                          "--prefix=#{prefix}",
+                          "--enable-static"
```
Comment (dunn) [inline Library/Formula/libassuan.rb]: This probably warrants a `revision 1` cc @DomT4

**Comment by DomT4** [comment]
Heh, I hadn't forgotten about this @jeroenooms :smiley:; Just caught up lately on finishing up my current Uni module. The other reason I hadn't moved on this just yet is because the recent GPGME build kept failing to build (See #43307). Hopefully the error I saw there doesn't keep reproducing.

I've no objection here.
My gut feeling was to make it optional, but providing everything works, `¯\_(ツ)_/¯`.

File: Library/Formula/libassuan.rb (hunk 1 above)
Comment (DomT4) [inline Library/Formula/libassuan.rb]: I think it'd be okay, but all the bottles are `cellar :any` so we can play it better safe than sorry.

**Comment by jeroenooms** [comment]
It would be really nice if `--enable-static` is the default config so that the static libraries are included with the bottles, as they are for libgpgme. The same is done for most linux distributions.

**Comment by DomT4** [comment]
I don't really object enough to making it the default to stop it being so. If people see problems we may have to look at it again, but there are things we build both for so it's not unprecedented. Bump the revisions on `libassuan` and `libgpg-error` and I'll merge this. It'd be nice to use this download link for GPGME as well:

```
https://gnupg.org/ftp/gcrypt/gpgme/gpgme-1.6.0.tar.bz2
```

**Comment by jeroenooms** [comment]
OK added the mirror URL. Where do I bump the revision? I see no such field in the recipe?

**Comment by DomT4** [comment]
It would be under the `sha256` line if it already existed. The default revision is `0` which we don't write. So in this case, just:

```diff
 url "blah"
 mirror "blah"
 sha256 "blah"
+revision 1
```

**Comment by jeroenooms** [comment]
ok done, thank you!

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/c2c71a24c1869e65d67fea997e3205036e7733df, https://github.com/Homebrew/homebrew/commit/61ddb9fcc5fae7bd3ee82831b5385ea737802d4a & https://github.com/Homebrew/homebrew/commit/a6a93d08a6f810fdcd8d531ffca60a724fb9a33c. Thank you for your contribution to Homebrew @jeroenooms!

**Pull request closed without merging**